packages/rs-dpp/src/document/state_transition/documents_batch_transition/document_transition/document_in_state_transition.rs

```diff
@@ -0,0 +1,510 @@
+use crate::data_contract::DataContract;
+use crate::identifier::Identifier;
+use crate::metadata::Metadata;
+use crate::prelude::{Revision, TimestampMillis};
+use crate::util::cbor_value::{CborCanonicalMap, FieldType};
+use crate::util::deserializer::SplitProtocolVersionOutcome;
+use crate::util::hash::hash;
+use crate::util::json_value::JsonValueExt;
+use crate::util::json_value::ReplaceWith;
+use crate::util::{cbor_value, deserializer};
+use crate::ProtocolError;
+use ciborium::Value as CborValue;
+use integer_encoding::VarInt;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use std::collections::HashSet;
+use std::convert::TryInto;
+
+pub mod property_names {
+    pub const PROTOCOL_VERSION: &str = "$protocolVersion";
+    pub const ID: &str = "$id";
+    pub const DOCUMENT_TYPE: &str = "$type";
+    pub const REVISION: &str = "$revision";
+    pub const DATA_CONTRACT_ID: &str = "$dataContractId";
+    pub const OWNER_ID: &str = "$ownerId";
+    pub const CREATED_AT: &str = "$createdAt";
+    pub const UPDATED_AT: &str = "$updatedAt";
+}
+
+pub const IDENTIFIER_FIELDS: [&str; 3] = [
+    property_names::ID,
+    property_names::DATA_CONTRACT_ID,
+    property_names::OWNER_ID,
+];
+
+/// The document object represents the data provided by the platform in response to a query.
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct DocumentInStateTransition {
```

Comment:
This document is not part of state transition. State transition contains other DocumentCreateTransition and so on.. This struct is mostly used by developers in his application and call it `DocumentInStateTransition` kinda confusing